### ISSUE TITLE
Tag Differences in previous years?  <smses> vs <allsms>

### ISSUE THREAD

**Issue by happyendings59** [MrMDavidson/SmsAndMmsMerger#3]
Did the backup tool change tags around 2015?

i got this error when processing files from 2015

It looks like the command encountered an error while processing the file "sms-20150624212205.xml". The error message says "Expected to encounter <smses> instead found <allsms>. Aborting." It seems that this file has a different XML structure compared to the other files in the directory, causing the program to fail.

**Comment by MrMDavidson** [comment]
> Did the backup tool change tags around 2015?
> 
> i got this error when processing files from 2015
> 
> It looks like the command encountered an error while processing the file "sms-20150624212205.xml". The error message says "Expected to encounter instead found . Aborting." It seems that this file has a different XML structure compared to the other files in the directory, causing the program to fail.

I'm honestly not sure - I've only got files dating back to ~2021 or so. If you're providing a mocked up sample I can try and add support but cannot guarantee the success of it or when I'd be able to.

**Comment by happyendings59** [comment]
i sent you a sample via email.

**Comment by MrMDavidson** [comment]
Pull latest - it should now support `<allsms>` as a top level tag. Thanks for sending through a sample file.

**Comment by happyendings59** [comment]
Will do soon.  I think what would be perfect is if it could convert the old
style messages to the new format.  I asked the vendor for info if there's
some way to convert.  I'll keep you posted.

**Comment by MrMDavidson** [comment]
Yup, so if you use the latest it'll support reading the older `<allsms>` files but produce the newer `<smses>` style file.

**Comment by happyendings59** [comment]
Have you actually tried the output to import on an android phone yet?
I'm...  hesitant to do so on my main phone and may either purchase a cheap
one or use bluestacks/some kinda simulator first (never tried before).

I wouldn't doubt an import would take all day and i'd really be... bleh if
somehow on import it messed up all my SMS and need to somehow go from a
backup.

I'm thinking maybe one of these:

https://www.amazon.com/TracFone-View-Prepaid-Smartphone-Locked/dp/B09XFJY4N4/ref=sr_1_2?keywords=cheap+android+recent+phone&qid=1677350493&refinements=p_n_feature_nine_browse-bin%3A23732472011&rnid=4972967011&s=wireless&sr=1-2

**Comment by happyendings59** [comment]
oops... that is a tracFone... maybe this one?

https://www.amazon.com/dp/B085TCRFST?tag=georiot-us-default-20&th=1&ascsubtag=dcw-us-4903441537345116000-20&geniuslink=true

**Comment by happyendings59** [comment]
it might be working and i tried doing everything by script using merging of newest to oldest (script below that might work but still getting the error about the invalid characters because the XML reader in that 2nd error report)

https://github.com/MrMDavidson/SmsAndMmsMerger/issues/2

I think by default the program uses this type of date format

The date pattern used in the filename "sms-20190222004032.xml" appears to be "YYYYMMDDHHMMSS".

i wonder if we should just ignore special characters in the old backups as a "oh well" solution?
